### PR TITLE
Add API to add features to existing biomes

### PR DESCRIPTION
**Pull request by modmuss50**
Also updated the module based of the old wiki page for adding ores.

### CODE REVIEW

**Review by shartte — APPROVED** [review]

**Review by i509VCB — APPROVED** [review]

**Review by Player3324 — APPROVED** [review]

File: fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biome/InternalBiomeData.java
Comment (liach) [inline fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biome/InternalBiomeData.java:156]: Hmm, do we have rule on whether put assignments in individual statements? Though this type of call is indeed better performance wise

**Review by liach — COMMENTED** [review]
My main concern is how this mutable setup will work with Mojang DFU's immutability contract (i.e. pretty much any object is immutable and mutation can cause unexpected behaviors)

**Comment by liach** [comment]
Another note: Does that `List<List<Supplier<ConfiguredFeature<?, ?>>>>` get sent to a codec? If it does, we need to watch out for codec serialization failure shall the codec be used to encode it over network/to disk.

File: fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biome/InternalBiomeData.java
Comment (modmuss50) [reply fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biome/InternalBiomeData.java:156]: I did this to save 2 lines of code (new vartible, add to map, reassign exisitng var) its not a super clear way of doing thing, but I think its fine here

**Comment by modmuss50** [comment]
Yes, I think this does get saved with the codec, im not aware of any issues this may cause?

**Comment by shartte** [comment]
I think it uses a registry codec and not a hardcoded map or similar, so should be fine (?)

**Comment by modmuss50** [comment]
Yes, you should ideally register everything you add here, we might need a seprate api to aid with that but its not as critical as this is.

**Review by Juuxel — APPROVED** [review]
> we might need a seprate api to aid with that

You can register them to `BuiltinRegistries.CONFIGURED_FEATURE` just fine with a simple register call.

**Review by liach — APPROVED** [review]
Yeah guess this is safe. We can revisit if we find this unsafe somehow.

**Review by YanisBft — APPROVED** [review]

**Review by Mysterious-Dev — APPROVED** [review]

**Comment by modmuss50** [comment]
This is now waiting on the biome API to be fixed, and I expect I will merge it all at once.

**Review by shedaniel — APPROVED** [review]
i cant bikeshed on a pr that i need immediately

**Comment by Mysterious-Dev** [comment]
This look good for a merge

**Comment by Shnupbups** [comment]
haha yes please merge i need

**Comment by modmuss50** [comment]
Its ready to be merged into a module that no longer exists...

Slight issue there.

File: fabric-biomes-v1/src/testmod/java/net/fabricmc/fabric/test/biome/FabricBiomeTest.java
Comment (shedaniel) [inline fabric-biomes-v1/src/testmod/java/net/fabricmc/fabric/test/biome/FabricBiomeTest.java:48]: consider using something like <https://github.com/shedaniel/cloth-api/blob/1.16.2/cloth-dynamic-registry-api-v1/src/main/java/me/shedaniel/cloth/mixin/dynamic/registry/MixinDynamicRegistryManager.java>

this wont work for datapack biomes, also that requires moving the server entrypoint earlier.

File: fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biome/InternalBiomeData.java
Comment (shedaniel) [inline fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biome/InternalBiomeData.java:144]: I wish this can take a Supplier of the feature instead so I can actually just load the feature from jsons.

Comment (yinisyang) [reply fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biome/InternalBiomeData.java:144]: +1

**Comment by MarcusElg** [comment]
A defiantly needed addition to the api!

**Review by Yoghurt4C — APPROVED** [review]
therealfartfriend sent me here

**Comment by CheaterCodes** [comment]
Please look at #1036 for a 1.16.2 version of this.

**Comment by modmuss50** [comment]
Closing as #1036 superseeds this.

**Pull request closed without merging**
